### PR TITLE
Service discovery with SDK v2

### PR DESCRIPTION
**Pull request by dstepanov**

### CODE REVIEW

**Comment by dstepanov** [comment]
I left the previous route53 implementation untouched, it can be deleted if this is merged and rote53 us removed from parameter store dependency.

**Comment by graemerocher** [comment]
@sdelamo can you take a look

File: aws-service-discovery/src/main/java/io/micronaut/discovery/aws/servicediscovery/registration/AwsServiceDiscoveryRegistrationClient.java [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+            // try the metadata
+            if (instance.getMetadata().contains("instanceId")) {
+                opt = Optional.of(instance.getMetadata().asMap().get("instanceId"));
+            } else {
+                LOG.error("Cannot determine the instance ID. Are you sure you are running on AWS EC2?");
+            }
+        }
+
+        opt.ifPresent(instanceId -> {
+            if (discoveryService != null && discoveryService.healthCheckConfig() != null) {
```
Comment (dstepanov) [inline aws-service-discovery/src/main/java/io/micronaut/discovery/aws/servicediscovery/registration/AwsServiceDiscoveryRegistrationClient.java:155]: Any idea what is `discoveryService` supposed to be? It's only set in tests but it looks like it should be provided from somewhere.

Comment (graemerocher) [reply aws-service-discovery/src/main/java/io/micronaut/discovery/aws/servicediscovery/registration/AwsServiceDiscoveryRegistrationClient.java:155]: good question

Comment (dstepanov) [reply aws-service-discovery/src/main/java/io/micronaut/discovery/aws/servicediscovery/registration/AwsServiceDiscoveryRegistrationClient.java:155]: I did more digging and it used to be set during the registration:
```
discoveryService = getService(route53AutoRegistrationConfiguration.getAwsServiceId());
```
it was removed in the initial version of this class.

**Review by sdelamo — CHANGES_REQUESTED** [review]
@dstepanov could you add documentation ?

**Comment by sdelamo** [comment]
Is this related to https://github.com/micronaut-projects/micronaut-aws/issues/905 ?

**Comment by dstepanov** [comment]
My goal is to migrate to SDK v2. I think I chose the correctly API

File: aws-service-discovery/src/main/java/io/micronaut/discovery/aws/servicediscovery/AwsServiceDiscoveryClientConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+     *
+     * @param awsServiceId service id
+     */
+    public void setAwsServiceId(String awsServiceId) {
+        this.awsServiceId = awsServiceId;
+    }
+
+    @Override
+    public ConnectionPoolConfiguration getConnectionPoolConfiguration() {
+        return null;
```
Comment (graemerocher) [inline aws-service-discovery/src/main/java/io/micronaut/discovery/aws/servicediscovery/AwsServiceDiscoveryClientConfiguration.java:120]: Is this supposed to return `null`?

**Review by graemerocher — APPROVED** [review]
Looks good, just one query

**Comment by graemerocher** [comment]
Ah yeah and documentation

**Comment by dstepanov** [comment]
@sdelamo Where is the existing documentation for the service discovery module?

File: aws-service-discovery/src/main/java/io/micronaut/discovery/aws/servicediscovery/AwsServiceDiscoveryClientConfiguration.java (hunk 2 above)
Comment (dstepanov) [reply aws-service-discovery/src/main/java/io/micronaut/discovery/aws/servicediscovery/AwsServiceDiscoveryClientConfiguration.java:120]: It's a copy of `Route53ClientDiscoveryConfiguration` and it looks like pooling is not supported for the client.

**Pull request merged**